### PR TITLE
Add API test

### PR DESCRIPTION
**Pull request by brinxmat**
# Reviewer instructions:
 - From a PO-point-of-view, the non-error features are most interesting as the errors are mostly boilerplate
 - Is the basic idea correct (see below)?
 - Is all the metadata (look at the JSON files) in place?
 - Is there anything I have overlooked, or that does not seem logical?
 - I am not expecting the Karate-test to work as is — it will need work to actually function I am guessing

# This adds four endpoints:
- Search for journals with parameters query and year
  - if year is unspecified, current year is assumed
- Search for publishers with parameters query and year
  -  if year is unspecified, current year is assumed
- direct URL for single journal in a given year
- direct URL for single publisher in a given year

### CODE REVIEW

**Review by terjehell — DISMISSED** [review]

File: test/test_data/journal_search_body.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "size": 1,
+  "searchString": "query=__QUERY_INPUT__&year=__YEAR_INPUT__page=1&results=10",
+  "processingTime": __PROCESSING_TIME__,
+  "firstRecord": 1,
+  "nextResults": null,
+  "hits": [
+    {
+      "id": "https://__CURRENT_ENVIRONMENT__.nva.unit.no/channel/journal/__YEAR_INPUT__/446885",
+      "name": "Sensors",
+      "dbhId": "446885",
```
Comment (garshol) [inline test/test_data/journal_search_body.json]: channelId?

File: test/api-test.feature [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+      | JSON_MEDIA_TYPE    | 'https://' + path + '/channel/journal/2019/' + existingJournal     | JOURNAL_YEAR_BODY     |
+      | JSON_LD_MEDIA_TYPE | 'https://' + path + '/channel/publisher?query=Taylor'              | PUBLISHER_SEARCH_BODY |
+      | JSON_MEDIA_TYPE    | 'https://' + path + '/channel/publisher?query=Taylor'              | PUBLISHER_SEARCH_BODY |
+      | JSON_LD_MEDIA_TYPE | 'https://' + path + '/channel/publisher/2019/' + existingPublisher | PUBLISHER_YEAR_BODY   |
+      | JSON_MEDIA_TYPE    | 'https://' + path + '/channel/publisher/2019/' + existingPublisher | PUBLISHER_YEAR_BODY   |
+
+  Scenario Outline: Search returns no more than ten results
+    * configure headers = { 'Accept': JSON_LD_MEDIA_TYPE, 'Authorization: Basic ' + token }
+    * contentType = responseHeaders['Content-Type'][0]
+    Given url <VALID_URL>
```
Comment (axthosarouris) [inline test/api-test.feature:231]: VALID_URL_WITH_START_PARAMETER

**Review by axthosarouris — DISMISSED** [review]

File: test/api-test.feature [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    When method get
+    Then status 200
+    And match response.hits.length < 11
```
Comment (sarahse) [inline test/api-test.feature]: er `<11` bedre enn `<=10`?

**Review by sarahse — DISMISSED** [review]

File: test/api-test.feature [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+      | 'https://' + path + '/channel/journal/2019/' + existingJournal     |
+      | 'https://' + path + '/channel/publisher?query=Taylor'              |
+      | 'https://' + path + '/channel/publisher/2019/' + existingPublisher |
+
+  Scenario Outline: Query with unacceptable method returns Not acceptable error
+    * configure headers = { 'Accept': JSON_LD_MEDIA_TYPE, 'Authorization: Basic ' + token }
+    * contentType = responseHeaders['Content-Type'][0]
+    Given url 'https://' + path + '/channel/journal?query=Sensors'
+    When method <METHOD>
+    Then status 406
```
Comment (axthosarouris) [inline test/api-test.feature]: From Mozilla's documentation I think this is a 405- Method-not-allowed. (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)

File: test/api-test.feature [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    And match response.detail == 'Your request cannot be processed because the supplied parameter(s) "not" cannot be understood'
+    And match response.instance == 'https://' + path + '/channel/journal?woolen=jumper'
+    And match response.requestId == '#notnull'
+
+  Scenario Outline: Request with bad content type returns Not Acceptable
+    * configure headers = { 'Accept': <UNACCEPTABLE_CONTENT_TYPE>, 'Authorization: Basic ' + token }
+    * contentType = responseHeaders['Content-Type'][0]
+    Given url <VALID_URL>
+    When method get
+    Then status 406
```
Comment (axthosarouris) [inline test/api-test.feature:192]: Should we also supply the acceptable types?

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**